src/common/shared.test.ts

```diff
@@ -90,4 +91,79 @@ describe("shared", () => {
       );
     });
   });
+
+  describe("getCodeForFailedResult", () => {
+    let mockIpc: RendererProcessIpc;
+
+    beforeEach(() => {
+      // @ts-expect-error partial implementation for testing
```

Comment:
`getCodeForFailedResult` requires the full type so while using `Partial` in this case would resolve the error, it would create new errors on each test that references this object. The `@ts-expect-error` is basically just an alternative to implementing all of the other functions that `RendererProcessIpc` uses. Another potential option would be to use `any`, but I figured a comment with an explanation would be the best option.

Reply:
@lucasfcosta have a look at https://github.com/elastic/synthetics-recorder/pull/143/commits/b772823cd88deee53b1fe207538910d8d27eb1bf - a direct cast didn't work because `RendererProcessIpc` doesn't overlap with my mock, so I had to cast to unknown first.